src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
```

Comment:
As I said before, no matter what happens, split, merge or other, PD has the finally region informations. 
So if we find a region which range covers the local region's start key, and version > local region, we can think the local region is stale .

If not stale, the local region is overlapped with the PD's, and this can't happen.